src/components/ha-textfield.ts

```diff
@@ -83,7 +83,7 @@ export class HaTextField extends TextFieldBase {
       }
 
       input {
-        text-align: var(--text-field-text-align);
+        text-align: var(--text-field-text-align, start);
```

Comment:
Somehow (voodoo), even if --text-field-text-align is not defined it defaults to right which puts the caret on the right even in English.